### ISSUE TITLE
Audio generation using PWM Block PCA9865

### ISSUE THREAD

**Issue by sriranjanr** [sparkfun/SparkFun_PWM_Block_for_Edison_CPP_Library#1]
Hello,

I want to generate audio using this PWM block. I need a PWM ouput at 2khz, 8khz frequency.Is it possible with this or should I use a PCA9835 (by desoldering the PCA9865 and soldering the 35 and making changes to the library appropriately.Please let me know.

**Comment by mhord** [comment]
I'm confused- do you need to modulate an 8kHz tone at 2kHz? If that's the case, you would need two blocks, since each one only has one frequency control. I'm not sure off the top of my head how accurate a signal you can get at those frequencies, either. You'd also need some external circuitry to strip off the higher harmonics and switch the frequency, and I'm not sure how you'd synchronize the two clocks.

In short, it's possible, but challenging.

**Comment by sriranjanr** [comment]
I read that the PCA9865 cant do a 2KZ PWM and neither can the PCA9835

**Comment by mhord** [comment]
Well, there's the answer, then!

You may consider other options for hardware. The [MiniGen(https://www.sparkfun.com/products/11420)] is a good, small form-factor signal generator that could do both 8 and 2kHz with respectable accuracy and good sinewave output.
